### PR TITLE
WebinarGeek #3835

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/webinargeek/sources/new-payment/new-payment.mjs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        id: data.id,
+        summary: `New payment with id ${data.id}`,
+        ts: Date.parse(data.created_at),
+      });
+    },
+    _setLastPaymentId(id) {
+      this.db.set("lastPaymentId", id);
+    },
+    _getLastPaymentId() {
+      this.db.get("lastPaymentId");
```
Comment (andrewjschuang) [inline components/webinargeek/sources/new-payment/new-payment.mjs:36]:
```suggestion
      return this.db.get("lastPaymentId");
```

File: components/webinargeek/sources/new-payment/new-payment.mjs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          page,
+          per_page: 100,
+        },
+      });
+
+      payments.reverse().forEach(this.emitEvent);
+
+      if (
+        payments.length < 100 ||
+        payments.filter((payment) => payment.id === lastPaymentId)
```
Comment (andrewjschuang) [inline components/webinargeek/sources/new-payment/new-payment.mjs:67]: Could you please explain to me this condition?

File: components/webinargeek/sources/new-registration/new-registration.mjs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          page,
+          per_page: 100,
+        },
+      });
+
+      subscriptions.reverse().forEach(this.emitEvent);
+
+      if (
+        subscriptions.length < 100 ||
+        subscriptions.filter((subscription) => subscription.id === lastSubscriptionId)
```
Comment (andrewjschuang) [inline components/webinargeek/sources/new-registration/new-registration.mjs:67]: Could you please explain to me this condition?

File: components/webinargeek/sources/new-registration/new-registration.mjs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        id: data.id,
+        summary: `New registration with id ${data.id}`,
+        ts: Date.parse(data.created_at),
+      });
+    },
+    _setLastSubscriptionId(id) {
+      this.db.set("lastSubscriptionId", id);
+    },
+    _getLastSubscriptionId() {
+      this.db.get("lastSubscriptionId");
```
Comment (andrewjschuang) [inline components/webinargeek/sources/new-registration/new-registration.mjs:36]:
```suggestion
      return this.db.get("lastSubscriptionId");
```

**Review by andrewjschuang — COMMENTED** [review]
@lcaresia thanks for the contribution!

I've left some comments below.

File: components/webinargeek/sources/new-payment/new-payment.mjs (hunk 2 above)
Comment (lcaresia) [reply components/webinargeek/sources/new-payment/new-payment.mjs:67]: Yes, default return is ordered by `created_at desc`, so I save the last ID emited, and in the next run if I find this ID I will stop to don't do unnecessary requests.

Comment (andrewjschuang) [reply components/webinargeek/sources/new-payment/new-payment.mjs:67]: Ahh ok, thanks!

**Review by andrewjschuang — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed, This is ready for release.

**Pull request merged**